[milestone/12.0.2] Milestone 12.0.2

felix-schwarz opened the pull request:
## Description
Aggregating all changes for release 12.0.2.

https://github.com/owncloud/ios-app/issues/1231

## Related Issues

- [x] solves #1219
- [x] fixes #1228 
- [x] fixes #1223
- [x] fixed #1225 by adding `Recents` sidebar item and `:recent` search shortcut

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] Added an issue with details about all relevant changes in the [**iOS documentation repository**](https://github.com/owncloud/docs-client-ios-app/issues).
- [ ] I have read the [**CONTRIBUTING**](https://github.com/owncloud/ios-app/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Added changelog files for the fixed issues in folder changelog/unreleased

felix-schwarz committed:
- add OAuth2/OIDC "state" parameter support via SDK update

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/owncloud/ios-app?pullRequest=1230) <br/>All committers have signed the CLA.

felix-schwarz committed:
- FileProvider fixes:
	- fix broken thumbnails for VFS item IDs
	- fix #1228

felix-schwarz committed:
- bump version to 12.0.2 and build to 269

hosy committed:
implemented recents item in Quick Access section

felix-schwarz committed:
- Add ":recent" shortcut
- Change "Recents" sidebar item search to use ":recent" instead of ":1w", replicating the app behaviour in version 11

felix-schwarz committed:
- add "Recently used" as quick choice for search

jesmrec commented:
## QA

### Changelog items

- [X] `state` sent in OIDC authentication process 
- [X] attach oC files to 3rd party apps via Files app (downloaded and not downloaded). Tested with `Mail` app (`Whatsapp` not available in testing devices)
- [ ] Browse file through Files App in iPad
- [X] Recent section in Quick access
- [X] `:recent` shortcut in search bar

### Testing

- [X] Smoke test on oC10
- [X] Smoke test on oCIS

### Update

- [ ] Update from 12.0
- [ ] Update from 12.0.1

felix-schwarz committed:
- show "Recents" quick access item also after logging in, logging out and logging in again

jesmrec commented:
`12.0.2` approved from my side

Approved commits:

app: `bf2d0afb`
sdk: `e301f3d`

Xcode version: `14.3.1 (14E300c)`

Devices used:

IPhoneXR iOS 16.5
iPadPro iPadOS 16.5
iPadAir iPadOS 15.3

hosy committed:
- added changelog entries
- added in-app release notes

hosy committed:
Calens changelog updated

hosy committed:
updated App Store release notes

hosy committed:
Merge branch 'milestone/12.0.2' of https://github.com/owncloud/ios-app into milestone/12.0.2

hosy committed:
updated build number
